webapp/src/main/java/com/box/l10n/mojito/rest/repository/BranchStatisticWS.java

```diff
@@ -52,41 +44,70 @@ public class BranchStatisticWS {
-                .and(ifParamNotNull(search(search)))
-                .and(ifParamNotNull(deletedEquals(deleted)))
-                .and(ifParamNotNull(empty(empty)))
-                .and(ifParamNotNull(createdBefore(createdBefore)))
-                .and(ifParamNotNull(createdAfter(createdAfter)))
-                .and(totalCountLessThanOrEqualsTo(totalCountLte)),
-            pageable);
+    // Retrieve all the Branch Statistic Ids without the totalCountLte filter:
+    Page<Long> branchStatisticIds = sparseBranchStatisticRepository.findAllWithIdOnly(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    ifParamNotNull(createdBefore(createdBefore))).and(
+                    ifParamNotNull(createdAfter(createdAfter)))
+            , pageable);
 
-    // 2. Hydrate BranchStatistic entities for JSON response using eager loading to avoid N+1
-    // queries
-    List<BranchStatistic> branchStatistics =
-        branchStatisticRepository.findByIdIn(branchStatisticIds.getContent(), pageable.getSort());
+    // Retrieve the Branch Statistic Ids which have TmTextUnit count less than the 'totalCountLte'
+    Page<Long> branchStatisticIdsWithLessThan = sparseBranchStatisticRepository.findAllWithIdOnly(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    ifParamNotNull(createdBefore(createdBefore))).and(
+                    ifParamNotNull(createdAfter(createdAfter))).and(
+                    totalCountLessThanOrEqualsTo(totalCountLte))
+            , pageable);
 
+    // Retrieve the branch statistics where the total text unit count is greater than the limit,
+    // by filtering the branchStatisticIds less than the limit.
+    Page<BranchStatistic> branchStatisticsWithoutTextUnits = branchStatisticRepository.findAll(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    totalCountGreaterThan(totalCountLte))
+            , pageable);
```

Comment:
Same comment from above applies here - it's not safe to use pageable again - this query should use as input the IDs from the very first query and the count desired.

However, we shouldn't go to the DB for this set of entries - we can take branchStatisticIds, eliminate IDs  from branchStatisticIdsWithLessThan - and the resulting IDs are exactly what we're looking for.